In [ ]:
import pandas as pd
import tensorflow as tf
import seaborn as sns
df = pd.read_csv('/content/phishing_url_dataset.csv')
print('Total rows: {}'.format(df.shape[0]))
print('Total columns: {}'.format(df.shape[1]))
data = df.values
train = data[:,1:3]
l = []
for i in train:
    if(i[0] <= 40):
        l.append("Small")
        if(i[1] == True):
          l.append("PS")
        else:
          l.append("FS")
    elif(i[0] <= 60):
        l.append("Medium")
        if(i[1] == True):
          l.append("PM")
        else:
          l.append("FM")
    else:
        l.append("Large")
        if(i[1] == True):
          l.append("PL")
        else:
          l.append("FL")
print((l.count("FS")/l.count("Small")))
print((l.count("FM")/l.count("Medium")))
print((l.count("FL")/l.count("Large")))

sns.set_style('darkgrid')
sns.countplot(l)


FileNotFoundError: ignored

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd
import tensorflow as tf
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import mean_squared_error, r2_score
from sklearn import metrics
df = pd.read_csv('/content/phishing_url_dataset.csv')
print('Total rows: {}'.format(df.shape[0]))
print('Total columns: {}'.format(df.shape[1]))
df['numeric_character_count'] = df['url'].apply(lambda x: sum(c.isdigit() for c in x))# urls = data['url']
urls = df['url']
x = df.drop(['url','phishing_attempt'], axis=1)
y = df['phishing_attempt']
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.33, random_state=42)
rf_classifier = RandomForestClassifier(n_estimators=75, criterion='entropy', max_depth=10,min_samples_split=2, min_samples_leaf=1, min_weight_fraction_leaf=0.0, max_features=.75, max_leaf_nodes=None,bootstrap=True, oob_score=False, n_jobs=1, random_state=None, verbose=5, warm_start=False,class_weight=None)
rf_classifier.fit(x_train, y_train)
y_pred = rf_classifier.predict(x_test)
print("Model accuracy = ", metrics.accuracy_score(y_test, y_pred))
data_with_predictions = y_test.to_frame('actual_value')
data_with_predictions.insert(0, 'predicted_value', y_pred)
data_with_predictions = urls.to_frame().join(data_with_predictions, how='inner')
data_with_predictions.head()
pd.Series(rf_classifier.feature_importances_, index = np.array(x_train.columns))
# y_train.head()


In [ ]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np
df = pd.read_csv('/content/phishing_url_dataset.csv')
%matplotlib inline
for col in df.columns:
    unique_value_list = df[col].unique()
    if len(unique_value_list) > 10:
        print(f'{col} has {df[col].nunique()} unique values')
    else:
        print(f'{col} contains:\t\t\t{unique_value_list}')
from sklearn.model_selection import train_test_split
from sklearn.model_selection import KFold
from xgboost import XGBClassifier
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score
def binary_classification_accuracy(actual, pred):

    print(f'Confusion matrix: \n{confusion_matrix(actual, pred)}')
    print(f'Accuracy score: \n{accuracy_score(actual, pred)}')
    print(f'Classification report: \n{classification_report(actual, pred)}')

features = df.drop(['url','phishing_attempt'], axis=1)
target = df['phishing_attempt']
folds = KFold(n_splits=4, shuffle=True, random_state=42)

train_index_list = list()
validation_index_list = list()

for fold, (train_idx, validation_idx) in enumerate(folds.split(features, target)):

    model = XGBClassifier()
    model.fit(np.array(features)[train_idx,:], np.array(target)[train_idx])
    predicted_values = model.predict(np.array(features)[validation_idx,:])
    print(f'==== FOLD {fold+1} ====')
    binary_classification_accuracy(np.array(target)[validation_idx], predicted_values)

FileNotFoundError: ignored

In [ ]:
import pandas as pd

import numpy as np
import re
import socket
df = pd.read_csv('/content/phishing_url_dataset.csv')
data = df.values
train = data[:,0:1]
urls = df['url']
print(train[0][0])
Igiveup = pd.Series(["nobell[.]it/70ffb52d079109dca5664cce6f31737378",  "www[.]dghjdgf[.]com/paypal[.]co[.]uk/cycgi-bin", "serviciosbys[.]com/paypal[.]cgi[.]bin[.]get-in"])
Igiveup.str.match(pat = '([A-Z]i.)')
print(Igiveup)
df.head()
# for k in train:
  # IP_addres = socket.gethostbyname((k[0]))
  # print("Computer IP Address is:" + IP_addres)
#23.22.39.120
#nobell[.]it/70ffb52d079109dca5664cce6f317373782/login[.]SkyPe[.]com/en/cgi-bin/verification/login/70ffb52d079109dca5664cce6f317373/index[.]php?cmd=_profile-ach&outdated_page_tmpl=p/gen/failed-to-load&nav=0[.]5[.]1&login_access=1322408526

FileNotFoundError: ignored

In [ ]:
df = pd.read_csv('/content/phishing_url_dataset.csv')
urls = df['url']
SeriesCircuit = pd.Series(urls)
inx = []
for i in range(len(SeriesCircuit)):
  inx.append(i)
SeriesCircuit.index = inx
SeriesCircuit.str.match(pat = '.com')
print(SeriesCircuit)


NameError: ignored

In [ ]:
# importing pandas as pd
import pandas as pd

# importing re for regular expressions
import re

# Creating the Series
sr = pd.Series(['New_York', 'Lisbon', 'Tokyo', 'Paris', 'Munich'])

# Creating the index
idx = ['City 1', 'City 2', 'City 3', 'City 4', 'City 5']

# set the index
sr.index = idx

# Print the series
print(sr)
# match either 'Tokyo' or 'Paris'
result = sr.str.match(pat = '(Tokyo)|(Paris)')

# print the result
print(result)




City 1    New_York
City 2      Lisbon
City 3       Tokyo
City 4       Paris
City 5      Munich
dtype: object
City 1    False
City 2    False
City 3     True
City 4     True
City 5    False
dtype: bool


In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from keras.models import Model
from keras.layers import LSTM, Activation, Dense, Dropout, Input, Embedding
from keras.optimizers import RMSprop
from keras.preprocessing.text import Tokenizer
from keras.preprocessing import sequence
from tensorflow.keras.utils import to_categorical
from keras.callbacks import EarlyStopping
%matplotlib inline

df = pd.read_csv('/content/phishing_url_dataset.csv')
killo = df.drop(['url_character_count','phishing_attempt'], axis=1)
data = df.values
x = data[:,0:1]
X = []
for k in x:
  X.append(k[0])
y = data[:,2:3]
Y = []
for ky in y:
  Y.append(ky[0])
df = df.drop(['url_character_count'],axis=1)
X_train,X_test,Y_train,Y_test = train_test_split(X,y,test_size=0.15)
max_words = 1000
max_len = 150
tok = Tokenizer(num_words=max_words)
tok.fit_on_texts(X_train)
sequences = tok.texts_to_sequences(X_train)
sequences_matrix = sequence.pad_sequences(sequences,maxlen=max_len)
def RNN():
    inputs = Input(name='inputs',shape=[max_len])
    layer = Embedding(max_words,50,input_length=max_len)(inputs)
    layer = LSTM(64)(layer)
    layer = Dense(256,name='FC1')(layer)
    layer = Activation('relu')(layer)
    layer = Dropout(0.5)(layer)
    layer = Dense(1,name='out_layer')(layer)
    layer = Activation('sigmoid')(layer)
    model = Model(inputs=inputs,outputs=layer)
    return model
model = RNN()
model.summary()
model.compile(loss='binary_crossentropy',optimizer=RMSprop(),metrics=['accuracy'])
sequences_matrix = np.asarray(sequences_matrix).astype('float32')
Y_train = np.asarray(Y_train).astype('float32')

model.fit(sequences_matrix,Y_train,batch_size=32,epochs=10,validation_split=0.2,callbacks=[EarlyStopping(monitor='val_loss',min_delta=0.0001)])
test_sequences = tok.texts_to_sequences(X_test)
test_sequences_matrix = sequence.pad_sequences(test_sequences,maxlen=max_len)
test_sequences_matrix = np.asarray(test_sequences_matrix).astype('float32')
Y_test = np.asarray(Y_test).astype('float32')
accr = model.evaluate(test_sequences_matrix,Y_test)
print('Test set\n  Loss: {:0.3f}\n  Accuracy: {:0.3f}'.format(accr[0],accr[1]))


# max_features = 6000

# maxlen = 130

# embed_size = 128
# model = Sequential()
# model.add(Embedding(max_features, embed_size))
# model.add(Bidirectional(LSTM(32, return_sequences = True)))
# model.add(GlobalMaxPool1D())
# model.add(Dense(20, activation="relu"))
# model.add(Dropout(0.05))
# model.add(Dense(1, activation="sigmoid"))
# model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

# batch_size = 100
# epochs = 3
# model.fit(x,y, batch_size=batch_size, epochs=epochs, validation_split=0.2)

www[.]onlinepictureproof[.]com/hotshotsindiana
Model: "model_25"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
inputs (InputLayer)          [(None, 150)]             0         
_________________________________________________________________
embedding_25 (Embedding)     (None, 150, 50)           50000     
_________________________________________________________________
lstm_25 (LSTM)               (None, 64)                29440     
_________________________________________________________________
FC1 (Dense)                  (None, 256)               16640     
_________________________________________________________________
activation_50 (Activation)   (None, 256)               0         
_________________________________________________________________
dropout_25 (Dropout)         (None, 256)               0         
_________________________________________________________________
out_layer (

In [ ]:
#Use this code for the presentation
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from keras.models import Model
from keras.layers import LSTM, Activation, Dense, Dropout, Input, Embedding
from keras.optimizers import RMSprop
from keras.preprocessing.text import Tokenizer
from keras.preprocessing import sequence
from tensorflow.keras.utils import to_categorical
from keras.callbacks import EarlyStopping
%matplotlib inline

df = pd.read_csv('/content/phishing_url_dataset.csv')
killo = df.drop(['url_character_count','phishing_attempt'], axis=1)
data = df.values
x = data[:,0:1]
X = []
for k in x:
  X.append(k[0])
y = data[:,2:3]
Y = []
for ky in y:
  Y.append(ky[0])
df = df.drop(['url_character_count'],axis=1)
X_train,X_test,Y_train,Y_test = train_test_split(X,y,test_size=0.15)
max_words = 1000
max_len = 150
tok = Tokenizer(num_words=max_words)
tok.fit_on_texts(X_train)
sequences = tok.texts_to_sequences(X_train)
sequences_matrix = sequence.pad_sequences(sequences,maxlen=max_len)
def RNN():
    inputs = Input(name='inputs',shape=[max_len])
    layer = Embedding(max_words,50,input_length=max_len)(inputs)
    layer = LSTM(64)(layer)
    layer = Dense(256,name='FC1')(layer)
    layer = Activation('relu')(layer)
    layer = Dropout(0.5)(layer)
    layer = Dense(1,name='out_layer')(layer)
    layer = Activation('sigmoid')(layer)
    model = Model(inputs=inputs,outputs=layer)
    return model
model = RNN()
model.summary()
model.compile(loss='binary_crossentropy',optimizer=RMSprop(),metrics=['accuracy'])
sequences_matrix = np.asarray(sequences_matrix).astype('float32')
Y_train = np.asarray(Y_train).astype('float32')

model.fit(sequences_matrix,Y_train,batch_size=32,epochs=7,callbacks=[EarlyStopping(monitor='val_loss',min_delta=0.0001)])
test_sequences = tok.texts_to_sequences(X_test)
test_sequences_matrix = sequence.pad_sequences(test_sequences,maxlen=max_len)
test_sequences_matrix = np.asarray(test_sequences_matrix).astype('float32')
Y_test = np.asarray(Y_test).astype('float32')
accr = model.evaluate(test_sequences_matrix,Y_test)
print('Test set\n  Loss: {:0.3f}\n  Accuracy: {:0.3f}'.format(accr[0],accr[1]))



Model: "model_5"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
inputs (InputLayer)          [(None, 150)]             0         
_________________________________________________________________
embedding_5 (Embedding)      (None, 150, 50)           50000     
_________________________________________________________________
lstm_5 (LSTM)                (None, 64)                29440     
_________________________________________________________________
FC1 (Dense)                  (None, 256)               16640     
_________________________________________________________________
activation_10 (Activation)   (None, 256)               0         
_________________________________________________________________
dropout_5 (Dropout)          (None, 256)               0         
_________________________________________________________________
out_layer (Dense)            (None, 1)                 257 

Model: "model_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
inputs (InputLayer)          [(None, 150)]             0         
_________________________________________________________________
embedding_1 (Embedding)      (None, 150, 50)           50000     
_________________________________________________________________
lstm_1 (LSTM)                (None, 64)                29440     
_________________________________________________________________
FC1 (Dense)                  (None, 256)               16640     
_________________________________________________________________
activation_2 (Activation)    (None, 256)               0         
_________________________________________________________________
dropout_1 (Dropout)          (None, 256)               0         
_________________________________________________________________
out_layer (Dense)            (None, 1)                 257       
_________________________________________________________________
activation_3 (Activation)    (None, 1)                 0         
=================================================================
Total params: 96,337
Trainable params: 96,337
Non-trainable params: 0
_________________________________________________________________
Epoch 1/7
2551/2551 [==============================] - 249s 90ms/step - loss: 0.2213 - accuracy: 0.9093
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 2/7
2551/2551 [==============================] - 232s 91ms/step - loss: 0.1360 - accuracy: 0.9443
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 3/7
2551/2551 [==============================] - 230s 90ms/step - loss: 0.1339 - accuracy: 0.9454
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 4/7
2551/2551 [==============================] - 233s 91ms/step - loss: 0.1458 - accuracy: 0.9446
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 5/7
2551/2551 [==============================] - 235s 92ms/step - loss: 0.1314 - accuracy: 0.9478
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 6/7
2551/2551 [==============================] - 238s 93ms/step - loss: 0.1257 - accuracy: 0.9503
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 7/7
2551/2551 [==============================] - 236s 93ms/step - loss: 0.1248 - accuracy: 0.9496
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
451/451 [==============================] - 11s 21ms/step - loss: 0.1224 - accuracy: 0.9512
Test set
  Loss: 0.122
  Accuracy: 0.951

Model: "model_5"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
inputs (InputLayer)          [(None, 150)]             0         
_________________________________________________________________
embedding_5 (Embedding)      (None, 150, 50)           50000     
_________________________________________________________________
lstm_5 (LSTM)                (None, 64)                29440     
_________________________________________________________________
FC1 (Dense)                  (None, 256)               16640     
_________________________________________________________________
activation_10 (Activation)   (None, 256)               0         
_________________________________________________________________
dropout_5 (Dropout)          (None, 256)               0         
_________________________________________________________________
out_layer (Dense)            (None, 1)                 257       
_________________________________________________________________
activation_11 (Activation)   (None, 1)                 0         
=================================================================
Total params: 96,337
Trainable params: 96,337
Non-trainable params: 0
_________________________________________________________________
Epoch 1/7
1276/1276 [==============================] - 157s 121ms/step - loss: 0.2409 - accuracy: 0.9007
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 2/7
1276/1276 [==============================] - 155s 121ms/step - loss: 0.1524 - accuracy: 0.9424
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 3/7
1276/1276 [==============================] - 154s 121ms/step - loss: 0.1310 - accuracy: 0.9480
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 4/7
1276/1276 [==============================] - 152s 119ms/step - loss: 0.1278 - accuracy: 0.9489
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 5/7
1276/1276 [==============================] - 152s 119ms/step - loss: 0.1321 - accuracy: 0.9479
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 6/7
1276/1276 [==============================] - 154s 121ms/step - loss: 0.1197 - accuracy: 0.9518
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
Epoch 7/7
1276/1276 [==============================] - 154s 121ms/step - loss: 0.1185 - accuracy: 0.9524
WARNING:tensorflow:Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,accuracy
451/451 [==============================] - 11s 23ms/step - loss: 0.1359 - accuracy: 0.9451
Test set
  Loss: 0.136
  Accuracy: 0.945

In [ ]:
import pandas as pd
import tensorflow as tf
import seaborn as sns
df = pd.read_csv('/content/phishing_url_dataset.csv')
print('Total rows: {}'.format(df.shape[0]))
print('Total columns: {}'.format(df.shape[1]))
data = df.values
train = data[:,1:3]
l = []
for i in train:
    if(i[0] <= 40):
        l.append("Small")
        if(i[1] == True):
          l.append("PS")
        else:
          l.append("FS")
    elif(i[0] <= 60):
        l.append("Medium")
        if(i[1] == True):
          l.append("PM")
        else:
          l.append("FM")
    else:
        l.append("Large")
        if(i[1] == True):
          l.append("PL")
        else:
          l.append("FL")
print((l.count("FS")/l.count("Small")))
print((l.count("FM")/l.count("Medium")))
print((l.count("FL")/l.count("Large")))

sns.set_style('darkgrid')
sns.countplot(l)

Execution output from Jul 19, 2021 11:08 AM
	Stream
		Total rows: 47896
		Total columns: 3
		0.0029302288746337214
		0.0012549914432401596
		0.0016978569272562633
		/usr/local/lib/python3.7/dist-packages/seaborn/_decorators.py:43: FutureWarning: Pass the following variable as a keyword arg: x. From version 0.12, the only valid positional argument will be `data`, and passing other arguments without an explicit keyword will result in an error or misinterpretation.
		  FutureWarning
	text/plain
		<matplotlib.axes._subplots.AxesSubplot at 0x7f5d41343650>
		<Figure size 432x288 with 1 Axes>

Code cell <fJDXG769WhKp>
#%% [code]
import pandas as pd
import tensorflow as tf
import seaborn as sns
dataset = pd.read_csv('/content/phishing_url_dataset.csv')
values = dataset.values
round = 1
base75_0 = 0
base75_1 = 0
base75_2 = 0
base75_3 = 0
base75_4 = 0
base75_5 = 0
keyword
keyl0raise = 1
keyl1raise = 5/6
keyl2raise = 4/6
keyl3raise = 3/6
keyl4raise = 2/6
keyl5raise = 1/6
keyletters0()
keyletters1()
keyletters2()
keyletters3()
keyletters4()
keyletters5()
map0 = {0:a,1:b,2:c,3:d,4:e,5:f,6:g,7:h,8:i,9:j,10:k,11:l,12:m,13:n,14:o,15:p,16:q,17:r,18:s,19:t,20:u,21:v,22:w,23:x,24:y,25:z}
map1 =
for round in range (0,180383630700):
  if base75_0 in range (0,26):
    base75_0 = {0:a,1:b,2:c,3:d,4:e,5:f,6:g,7:h,8:i,9:j,10:k,11:l,12:m,13:n,14:o,15:p,16:q,17:r,18:s,19:t,20:u,21:v,22:w,23:x,24:y,25:z}
    keyletters0.append(base75_0)
    base75_0 = {a:0,b:1,c:2,d:3,e:4,f:5,g:6,h:7,i:8,j:9,k:10,l:11,m:12,n:13,o:14,p:15,q:16,r:17,s:18,t:19,u:20,v:21,w:22,x:23,y:24,z:25}


  elif base75_0 in range (26,52):
    base75_0 = base75_0 - 26
    base75_0 = {0:a,1:b,2:c,3:d,4:e,5:f,6:g,7:h,8:i,9:j,10:k,11:l,12:m,13:n,14:o,15:p,16:q,17:r,18:s,19:t,20:u,21:v,22:w,23:x,24:y,25:z}
    base75_0.upper()
    keyletters0.append(base75_0)
    base75_0.lower();
    base75_0 = {a:0,b:1,c:2,d:3,e:4,f:5,g:6,h:7,i:8,j:9,k:10,l:11,m:12,n:13,o:14,p:15,q:16,r:17,s:18,t:19,u:20,v:21,w:22,x:23,y:24,z:25}

  elif base75_0 in range (52,62):
    base75_0 = base75_0 - 52
    keyletters0.append(base75_0)
    base75_0 = base75_0 + 52

  else base75_0 > 62:
    base75_0 = {63:!,64:@,65:"#",66:$,67:%,68:^,69:&,70:*,71:(,72:),73::,74:\,75:/}
    keyletters0.append(base75_0)
    base75_0 = {!:63,@:64,"#":65,$:66,%:67,^:68,&:69,*:70,(:71,):72,::73,\:74,/:75}


if keyl0raise % 1:
  base75_0 = base75_0 + 1
if keyl1raise % 1:
  base75_1 = base75_1 + 1
if keyl2raise % 1:
  base75_2 = base75_2 + 1
if keyl3raise % 1:
  base75_3 = base75_3 + 1
if keyl4raise % 1:
  base75_4 = base75_4 + 1
if keyl5raise % 1:
  base75_5 = base75_5 + 1

keyl0raise = keyl0raise + 1/6
keyl1raise = keyl1raise + 1/6
keyl2raise = keyl2raise + 1/6
keyl3raise = keyl3raise + 1/6
keyl4raise = keyl4raise + 1/6
keyl5raise = keyl5raise + 1/6

round = round + 1
if round % 1000000000:
  print(round + "rounds completed")

  for round in range (26, 180383630700):